src/attr/_compat.py

```diff
@@ -85,11 +85,38 @@ def iteritems(d):
     def metadata_proxy(d):
         return types.MappingProxyType(dict(d))
 
-if PYPY:  # pragma: no cover
-    def set_closure_cell(cell, value):
-        cell.__setstate__((value,))
-else:
-    import ctypes
-    set_closure_cell = ctypes.pythonapi.PyCell_Set
-    set_closure_cell.argtypes = (ctypes.py_object, ctypes.py_object)
-    set_closure_cell.restype = ctypes.c_int
+
+def import_ctypes():  # pragma: nocover
+    """
+    Moved into a function for testability.
+    """
+    try:
+        import ctypes
+        return ctypes
+    except ImportError:
+        return None
+
+
+def nop(*args, **kw):  # pragma: nocover
```

Comment:
i propose triggering a warning, based on the assumption that the super fixup code would produce broken code if its actually a noop

so in case the actual fixer is triggered, the user should get a warning that his/her code no longer works

Reply:
@hynek well, a user adding gae support might be toughly surprised - even more so of that perhaps happens in a dependency

i meant "no longer" works in the sense that the new environment secretly breaks their neck and they get broken code over there as a result

in addition is an easy oversight for review, its easy to imply "yay more memory efficient" and easy to forget to implement "this is a breaking change that kills gae support" for users

Reply:
actually i believe its ok to go as far as directly erroring out since we cant correct it on gae and else it creates broken code - code with warning would still later on break on actual use

Reply:
Surprised by what exactly?  In 17.2 bare super() didn’t work for anybody.  It does work in 17.3 thru ctypes but it doesn’t work on GAE.  In 17.4 the behavior from 17.2 will be restored for GAE. There’s nothing we take away from them.  I’d just like to document it somewhere and be done with it.

Reply:
I have added a warning on Python 3 if ctypes is missing.

If you want any stronger measures, please make a more explicit case in the PR comments, thanks!

Reply:
it would be nice if it only triggered when setting a closure cell actually, so it isn't warning when using things that work